### DISCUSSION TITLE
Some suggestions for Maui project - master

### DISCUSSION THREAD

**Discussion by shawyunz** [rwecho/V2ex.Maui#11] [General]
Suggestions:
1. Keep either PageViewModel or ViewModel
2. Use Route in AppShell instead of command
3. Use native Dependency Injection instead of Volo.Abp.Autofac
4. Resx resources for localization (Maintainability)
5. Project folder structures

Reference project: https://github.com/dotnet-architecture/eshop-mobile-client

**Comment by shawyunz** [comment]
Append:
Aligning more than 10 tabs horizontally is not a good practice (in AppShell - MainFlyoutItem).
Options: Use TabView instead of TabPage. Or replace it with a dropdown list/menu like V2er.Android
Also a custom header can fix issue #6 to hide title when scroll.

**Reply by rwecho** [reply]
Where are the 10 tabs? here? https://github.com/rwecho/V2ex.Maui/blob/886a478403e75954377e71933e88229525c4cbac/src/V2ex.Maui/AppShell/AppShell.xaml#L28-L35

And for the custom header, we should do that.

I have referenced the issue from the MAUI project.  Hiding the title with animation is our goal like X(Twitter).

**Reply by shawyunz** [reply]
https://github.com/rwecho/V2ex.Maui/blob/886a478403e75954377e71933e88229525c4cbac/src/V2ex.Maui/AppShell/AppShell.xaml.cs#L45-L79

**Reply by rwecho** [reply]
yeah, I neglected here. We'd better create a .editorconfig file to constraint code style.

**Comment by rwecho** [comment]
Regarding 1, 2, and 5, I will study the eShop project first and then make adjustments.
3 and 4 will be fixed at some time in the future. Because it's not important to compare the other features like markdown, reply, click to redirect, etc.

**Reply by shawyunz** [reply]
I can do 125 first, any plan to merge your dev branch at some point?

**Reply by rwecho** [reply]
Sure, I have sent an invitation to you.

**Reply by rwecho** [reply]
Now, I develop at the develop branch, and for every feature, I'll create a branch like features/fix-AppShellStyles, and merge it to develop .

https://nvie.com/posts/a-successful-git-branching-model/ 

And we will publish at the master of release* branch.

**Comment by rwecho** [comment]
We should make the app more convenient than other apps, for example, we can integrate some v2ex extension functions in Chrome.  
Making this the same to others is the basic point.
